pkg/controller/versions/strategy_v2_4.go

```diff
@@ -274,10 +274,41 @@ func (v *versionStrategyV2_4) ValidateRequest(ctx context.Context, cl client.Cli
 			return admission.ValidationResponse(false, "a cluster-scoped SMCP may only be created by users with cluster-admin permissions")
 		}
 	}
-
+	hasPilotSecretName := ContainsPilotSecretNameField(smcp)
+	if hasPilotSecretName {
+		return admission.ValidationResponse(false, "SMCP does not allow PilotSecretName field")
+}
+	hasRootCAConfigMapName := ContainsRootCAConfigMapNameField(smcp)
+	if hasRootCAConfigMapName {
+	return admission.ValidationResponse(false, "SMCP does not allow RootCAConfigMapName field")
+}
 	return admission.ValidationResponse(true, "")
 }
 
+	func ContainsPilotSecretNameField(smcp metav1.Object) bool {
+		switch s := smcp.(type) {
+	case *v1.ServiceMeshControlPlane:
+		return false
+	case *v2.ServiceMeshControlPlane:
+		return s.Spec.IsPilotSecretNameEnabled()
```

Comment:
I think it would be better if the code in this function was inlined here (i.e. remove the function and move the code here).